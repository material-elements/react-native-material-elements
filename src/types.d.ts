export * from './components/types';
export * from './types/index';
