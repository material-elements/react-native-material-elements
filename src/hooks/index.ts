export * from './useRestyle';
export * from './useThemedProps';
