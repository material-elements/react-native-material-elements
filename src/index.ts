export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
