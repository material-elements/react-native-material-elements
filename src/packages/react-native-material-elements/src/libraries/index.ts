export * from './themes';
export * from './style';
