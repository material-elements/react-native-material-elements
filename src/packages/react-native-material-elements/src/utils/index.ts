export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
