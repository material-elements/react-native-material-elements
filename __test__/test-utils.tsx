import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
