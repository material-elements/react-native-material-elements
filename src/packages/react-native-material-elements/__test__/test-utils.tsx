import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
