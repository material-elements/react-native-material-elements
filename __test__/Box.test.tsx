import React from 'react';
import { StyleSheet, View, ViewStyle } from 'react-native';
import { Box, Text } from '../src';
import { BaseStyles } from '../src/libraries/style/styleTypes';
import { render, waitFor } from './test-utils';

describe('Box Component', () => {
  const ref = React.createRef<View>();
  const mockTestId = 'mock_box';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', async () => {
    const { toJSON } = render(<Box sx={{ bg: 'red', r: 10, w: 20, h: 20 }} />);
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should match the snapshot with sx styles', async () => {
    const sx = { bg: 'red', p: 10, r: 5 };
    const { toJSON } = render(
      <Box sx={sx}>
        <View />
      </Box>,
    );
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should match the snapshot with custom styles', async () => {
    const customStyle = { backgroundColor: 'blue', padding: 20 };
    const { toJSON } = render(
      <Box style={customStyle}>
        <View />
      </Box>,
    );
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('renders without crashing', () => {
    const { getByTestId } = render(<Box testID={mockTestId} />);
    expect(getByTestId(mockTestId)).toBeTruthy();
  });

  it('should apply the custom styles when style prop is passed', () => {
    const { getByTestId } = render(
      <Box testID={mockTestId} style={{ backgroundColor: 'red', borderRadius: 10, margin: 10, padding: 20 }} />,
    );
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.backgroundColor).toEqual('red');
    expect(flattenedStyle.borderRadius).toEqual(10);
    expect(flattenedStyle.margin).toEqual(10);
    expect(flattenedStyle.padding).toEqual(20);
  });

  it('should apply the sx styles when sx prop is passed', () => {
    const sx: BaseStyles = { bg: 'red', r: 10, m: 10, p: 20 };
    const { getByTestId } = render(<Box testID={mockTestId} sx={sx} />);
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.backgroundColor).toEqual('red');
    expect(flattenedStyle.borderRadius).toEqual(10);
    expect(flattenedStyle.margin).toEqual(10);
    expect(flattenedStyle.padding).toEqual(20);
  });

  it('should combine the sx and styles props correctly', () => {
    const mockSx: BaseStyles = { bg: 'red', r: 10, m: 10, p: 20 };
    const mockStyles: ViewStyle = { backgroundColor: 'pink', borderRadius: 30 };

    const { getByTestId } = render(<Box testID={mockTestId} sx={mockSx} style={mockStyles} />);
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.backgroundColor).toEqual('pink');
    expect(flattenedStyle.borderRadius).toEqual(30);
    expect(flattenedStyle.margin).toEqual(10);
    expect(flattenedStyle.padding).toEqual(20);
  });

  it('renders children correctly', () => {
    const { getByText } = render(
      <Box testID={mockTestId}>
        <Text>Child Text</Text>
      </Box>,
    );
    expect(getByText('Child Text')).toBeTruthy();
  });

  it('should forward ref correctly', () => {
    render(
      <Box ref={ref}>
        <View />
      </Box>,
    );

    expect(ref.current).not.toBeNull();
    expect(ref.current).toBeInstanceOf(View);
  });

  it('should position child elements correctly', () => {
    const { getByTestId } = render(<Box testID={mockTestId} style={{ flexDirection: 'row', display: 'flex' }} />);
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.flexDirection).toEqual('row');
    expect(flattenedStyle.display).toEqual('flex');
  });

  it('should add the inline styles correctly', () => {
    const { getByTestId } = render(<Box testID={mockTestId} display="flex" flexDirection="row" />);
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.flexDirection).toEqual('row');
    expect(flattenedStyle.display).toEqual('flex');
  });

  it('should add the flex inline styles correctly', () => {
    const { getByTestId } = render(
      <Box testID={mockTestId} display="flex" flexDirection="row" justifyContent="center" alignItems="center" />,
    );
    const box = getByTestId(mockTestId);
    const flattenedStyle = StyleSheet.flatten(box.props.style);
    expect(flattenedStyle.flexDirection).toEqual('row');
    expect(flattenedStyle.display).toEqual('flex');
    expect(flattenedStyle.justifyContent).toEqual('center');
    expect(flattenedStyle.alignItems).toEqual('center');
  });

  it('should render the animated view', () => {
    const { toJSON } = render(<Box animatedView />);
    expect(toJSON()).toMatchSnapshot();
  });
});
