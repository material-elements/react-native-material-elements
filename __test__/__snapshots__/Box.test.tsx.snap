// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component should match the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      },
      {},
      undefined,
    ]
  }
/>
`;

exports[`Box Component should match the snapshot with custom styles 1`] = `
<View
  style={
    [
      {},
      {},
      {
        "backgroundColor": "blue",
        "padding": 20,
      },
    ]
  }
>
  <View />
</View>
`;

exports[`Box Component should match the snapshot with sx styles 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "red",
        "borderRadius": 5,
        "padding": 10,
      },
      {},
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`Box Component should render the animated view 1`] = `
<View
  collapsable={false}
  style={{}}
/>
`;
